website/app/components/doc/cards/card.gts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright IBM Corp. 2021, 2025
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { LinkTo } from '@ember/routing';
+
+import DocBadge from 'website/components/doc/badge';
+
+interface Status {
+  deprecated?: boolean;
+  updated?: boolean;
+  added?: boolean;
+}
+
+export interface DocCardsCardSignature {
+  Args: {
+    image?: string;
+    title: string;
+    caption?: string;
+    status?: Status;
+    route?: string;
+    model?: unknown;
+    href?: string;
+    layout?: 'vertical' | 'horizontal';
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+export default class DocCardsCard extends Component<DocCardsCardSignature> {
+  get classNames() {
+    const classes = ['doc-cards-card'];
+    const layout = this.args.layout ?? 'vertical';
+
+    // add a class based on the @layout argument, default = 'vertical'
+    classes.push(`doc-cards-card--layout-${layout}`);
+
+    return classes.join(' ');
+  }
+
+  <template>
+    <li class={{this.classNames}} ...attributes>
+      {{#if @route}}
+        <LinkTo
+          class="doc-cards-card__link"
+          @route={{@route}}
+          @model={{@model}}
+        >
+          <img
+            class="doc-cards-card__image"
+            src={{@image}}
+            alt=""
+            role="presentation"
+          />
+          <div class="doc-cards-card__details">
+            <p class="doc-cards-card__title">{{@title}}</p>
+            {{#if @status.deprecated}}
+              <DocBadge
+                class="doc-cards-card__badge"
+                @type="warning-inverted"
+                @size="medium"
+              >Deprecated</DocBadge>
+            {{else if @status.updated}}
+              <DocBadge
+                class="doc-cards-card__badge"
+                @type="neutral-inverted"
+                @size="medium"
+              >Updated</DocBadge>
+            {{else if @status.added}}
+              <DocBadge
+                class="doc-cards-card__badge"
+                @type="information-inverted"
+                @size="medium"
+              >Added</DocBadge>
+            {{/if}}
+            <p class="doc-cards-card_description">{{@caption}}</p>
+          </div>
```

Comment:
One nice thing of `gts` files is that one can define local (in the same file) components, to be used inside the `<template>` block. In this case, you can abstract away all this part, and avoid the duplication below.

Reply:
I didnt end up making a sub component, but I did switch to getters so the logic only happens in one place.